apps/asap-server/src/utils/sync-google-calendar.ts

```diff
@@ -0,0 +1,67 @@
+import { google, calendar_v3 as calendarV3, Auth } from 'googleapis';
+import logger from './logger';
+
+export type SyncCalendarFactory = (
+  syncToken: string | undefined,
+  syncEvent: (event: calendarV3.Schema$Event) => Promise<void>,
+  auth: Auth.GoogleAuth | Auth.OAuth2Client,
+) => (googleCalendarId: string) => Promise<string | undefined | null>;
```

Comment:
I know you are not really doing anything with those synced events in this PR apart from logging, but knowing that these events need to be saved later I would personally have `SyncCalendarFactory` return not just the token but also a full list of events that it fetched. 

This way we could separate the logic of saving events (or doing anything with them) from the syncing. I know that the function to sync events itself is separate (`syncEventFactory`), so yeah, you can mock it, test it, etc, but it would be much cleaner and easier to test the entire logic with the suggested approach, i think.